src/contracts/deposit/deposit.service.ts

```diff
@@ -257,21 +238,21 @@ export class DepositService {
    * Updates the cache deposited events
    * The last N blocks are not stored, in order to avoid storing reorganized blocks
    */
-  public async updateEventsCache(): Promise<void> {
+  public async updateEventsCache(): Promise<number> {
     const fetchTimeStart = performance.now();
 
     const [currentBlock, initialCache] = await Promise.all([
+      // TODO: check reorg, add finalized
```

Comment:
What is this TODO supposed to do? When will we address it?

Reply:
I had the idea of getting data not once in X blocks, but on each finalized block
I'll come back later and think about it, thanks for pointing it out.

Reply:
I think we currently fetch events for every new block we handle in handleNewBlock. We just update the cache for (blockNumber - lag)